proto_parser/src/proto_parser.cc

```diff
@@ -88,7 +101,42 @@ std::unique_ptr<ProtoValue> ProtoParser::CreateMessage(
 }
 
 void ProtoParser::PopulateComments(int field_loc, ProtoValue* message) {
-  // TODO(alexanderlin): Implement.
+  int comment_above_idx = field_loc - 1;
+
+  std::vector<std::string> comments_above;
+
+  // Acquire all comments.
+  while(comment_above_idx >= 0) {
+    std::string line = trim(lines_[comment_above_idx]);
```

Comment:
line is a confusing name here.

Try

std::string line_content = Trim(lines_[]);